lib/markdown2.py

```diff
@@ -391,8 +391,20 @@ def convert(self, text):
         text += "\n"
 
         rv = UnicodeWithAttrs(text)
-        if "toc" in self.extras:
+        if ("toc" in self.extras or "inline-toc" in self.extras):
+            # Generate TOC HTML as a property to be able to hijack it in "inline-toc" for subsctitution
+            # TODO (Tomas Hubelbauer): See about using that as a static method without a need for a throwaway instance.
             rv._toc = self._toc
+        if "inline-toc" in self.extras:
+            if self._toc[0] is None:
+                rv = UnicodeWithAttrs(rv.toc_html + text)
+            else:
+                (level, id, name) = self._toc[0]
+                # Need to use a regex and rely on the HTML structure, tracking the regex's `end()` across all the HTML transformations would be extremely unreliable
+                # TODO (Tomas Hubelbauer): Consider looser regex which allows for more attributes in order to to find heading even when more extras add attributes to it (future-proof)
+                pattern = r"\<h{} id=[\"\']{}[\"\']\>{}<\/h{}\>".format(level, id, re.escape(name), level)
+                text = re.sub(pattern, "\g<0>" + rv.toc_html, text)
```

Comment:
It would be nice to somehow check for the failure of the substitution and prepend the TOC HTML instead, but not sure what would be the best way to go about that.

Reply:
Oh and this probably doesn't deal well with multiple headings with the same text, I need to check for that.

- [ ] Check for that